main.tf

```diff
@@ -16,6 +16,10 @@ resource "aws_iam_user_login_profile" "default" {
   password_length         = var.password_length
   password_reset_required = var.password_reset_required
   depends_on              = [aws_iam_user.default]
+
+  lifecycle {
```

Comment:
Hm... Does adding the `lifecycle` block force a recreate of this resource? What does that do for existing IAM users that you're managing?

Reply:
After upgrade of AWS provider to 4.x (in our case - to 4.21, without any change to root module configuration or terraform vars) the resource was forced to recreate:
```
  # module.users["veronika_gnilitska"].aws_iam_user_login_profile.default[0] must be replaced
-/+ resource "aws_iam_user_login_profile" "default" {
      ~ encrypted_password      = "wcBMAwEF3O2PIAyrAQgAvoQ6klRbbQ705f0Pj6l9uf4C/9GOdVfiEqmBkendM3dKPRkbbcJVXRP80xaTkElae17OiS69QJVji3/sS3ru2TLnFyZcNtSMayu9sodi2VDbNOhN6FRywuv9OBkBcgDu7686pOZLG27H2zT7V18sBwYR6bDaUepnLtDLq" -> (known after apply)
      ~ id                      = "veronika.gnilitska" -> (known after apply)
      ~ key_fingerprint         = "92f19546ec0f7d7ca8f9d" -> (known after apply)
      + password                = (known after apply)
      ~ password_reset_required = false -> true # forces replacement
        # (3 unchanged attributes hidden)
    }
```
If you apply this plan you won't be able to login with the credentials used before the apply. To avoid this behaviour for existing users - lifecycle was used.

Reply:
👍 👍

Reply:
@Gowiem thank you! 💯